### DISCUSSION TITLE
A way to manually validate fields when call handleSubmit event

### DISCUSSION THREAD

**Discussion by bryanprimus** [react-hook-form/react-hook-form#8843] [Q&A]
I am creating a stepper where I store the value by separating the object for every step

```ts
type Inputs = {
  step1: {
    name: string;
    age: number;
  };
  step2: {
    email: string;
    password: string;
  };
};
```

Is there a way to manually validate the fields using `handleSubmit` ?

my current workaround is to use a button with `type="button"` instead of submit. 
where I will call, 

// first step
`trigger("step1")` this will validate all fields in step1 obj

// last step (2nd step)
and use `handleSubmit` here

The downside of the workaround approach is that user will lose a11y of keyboard nav, where user can't press enter when going the next step because the button is in `type=button`, and will result in many conditions in code to check for button type `"submit" or "button"`

My current imagination of how this is going to be is validating the form like this

```ts
form.handleSubmit(async (data) => {
   if(step === 1) {
     const isValidStep1 =  await trigger("step1")
     if(isValidStep1) { setStep((prevStep) => prevStep + 1) }
   } 
   // last step
   if(step === 2) {
      // submit logic here
   }   
}, (err) => {}, { isValidateAutomatically: false }: Options)
```

**Comment by Moshyfawn** [comment]
I'd recommend creating multiple different forms with `defaultValues` from previous steps instead

**Reply by bryanprimus** [reply]
do you have an example of how to achieve this.? all examples that I can find always store the data in a different state than react hook form.

**Reply by bryanprimus** [reply]
i always think that this can be done only with one `useForm`
